bigbluebutton-html5/imports/ui/components/chat/service.js

```diff
@@ -33,72 +33,81 @@ const getUser = (userId) => {
   return mapUser(user);
 };
 
-const mapMessage = (message) => {
+const mapGroupMessage = (message) => {
   const mappedMessage = {
     id: message._id,
     content: message.content,
-    time: message.fromTime, // + message.from_tz_offset,
+    time: message.timestamp,
     sender: null,
   };
 
-  if (message.type !== SYSTEM_CHAT_TYPE) {
-    mappedMessage.sender = getUser(message.fromUserId);
+  if (message.sender !== SYSTEM_CHAT_TYPE) {
+    mappedMessage.sender = getUser(message.sender);
   }
 
   return mappedMessage;
 };
 
-const reduceMessages = (previous, current) => {
+const reduceGroupMessages = (previous, current) => {
   const lastMessage = previous[previous.length - 1];
   const currentMessage = current;
-
   currentMessage.content = [{
-    id: current._id,
+    id: current.id,
     text: current.message,
-    time: current.fromTime,
+    time: current.timestamp,
   }];
-
-  if (!lastMessage || !currentMessage.type === SYSTEM_CHAT_TYPE) {
+  if (!lastMessage || !currentMessage.chatId === PUBLIC_GROUP_CHAT_ID) {
     return previous.concat(currentMessage);
   }
-
   // Check if the last message is from the same user and time discrepancy
   // between the two messages exceeds window and then group current message
   // with the last one
   const timeOfLastMessage = lastMessage.content[lastMessage.content.length - 1].time;
-  if (lastMessage.fromUserId === currentMessage.fromUserId
-    && (currentMessage.fromTime - timeOfLastMessage) <= GROUPING_MESSAGES_WINDOW) {
+  if (lastMessage.sender === currentMessage.sender
+    && (currentMessage.timestamp - timeOfLastMessage) <= GROUPING_MESSAGES_WINDOW) {
     lastMessage.content.push(currentMessage.content.pop());
     return previous;
   }
 
   return previous.concat(currentMessage);
 };
 
-const reduceAndMapMessages = messages =>
-  (messages.reduce(reduceMessages, []).map(mapMessage));
+const reduceAndMapGroupMessages = messages =>
+  (messages.reduce(reduceGroupMessages, []).map(mapGroupMessage));
 
-const getPublicMessages = () => {
-  const publicMessages = Chats.find({
-    type: { $in: [PUBLIC_CHAT_TYPE, SYSTEM_CHAT_TYPE] },
+const getPublicGroupMessages = () => {
+  const publicGroupMessages = GroupChatMsg.find({
+    chatId: PUBLIC_GROUP_CHAT_ID,
   }, {
-    sort: ['fromTime'],
+    sort: ['timestamp'],
   }).fetch();
 
-  return publicMessages;
+  return publicGroupMessages;
 };
 
-const getPrivateMessages = (userID) => {
-  const messages = Chats.find({
-    toUsername: { $ne: PUBLIC_CHAT_USERNAME },
-    $or: [
-      { toUserId: userID },
-      { fromUserId: userID },
-    ],
-  }, {
-    sort: ['fromTime'],
-  }).fetch();
-  return reduceAndMapMessages(messages);
+const getPrivateGroupMessages = (chatID) => {
+  const sender = getUser(Auth.userID);
+
+  const privateChat = GroupChat.findOne({
+    users: { $all: [chatID, sender.id] },
+    access: { $not: { $ne: PRIVATE_CHAT_TYPE } },
```

Comment:
Why are you looking for a double negative? Shouldn't it just use $eq?

Reply:
It was a problem we had, https://github.com/bigbluebutton/bigbluebutton/blob/v2.2-dev/bigbluebutton-html5/imports/ui/services/unread-messages/index.js#L41
I'll change to user `$eq`

Reply:
Ah. Interesting choice from minimongo